src/pkg/packager/lint/lint_test.go

```diff
@@ -9,136 +9,249 @@ import (
 	"errors"
 	"fmt"
 	"os"
-	"path/filepath"
 	"testing"
 
-	"github.com/defenseunicorns/zarf/src/config"
-	"github.com/defenseunicorns/zarf/src/pkg/packager/composer"
+	"github.com/defenseunicorns/zarf/src/pkg/variables"
 	"github.com/defenseunicorns/zarf/src/types"
 	goyaml "github.com/goccy/go-yaml"
 	"github.com/stretchr/testify/require"
 )
 
-const badZarfPackage = `
-kind: ZarfInitConfig
-metadata:
-  name: init
-  description: Testing bad yaml
-
-components:
-- name: first-test-component
-  import:
-    not-path: packages/distros/k3s
-- name: import-test
-  import:
-    path: 123123
-`
-
-const goodZarfPackage = `
-x-name: &name good-zarf-package
-
-kind: ZarfPackageConfig
-metadata:
-  name: *name
-  x-description: Testing good yaml with yaml extension
-
-components:
-  - name: baseline
-    required: true
-    x-foo: bar
-
-`
-
 func readAndUnmarshalYaml[T interface{}](t *testing.T, yamlString string) T {
 	t.Helper()
 	var unmarshalledYaml T
 	err := goyaml.Unmarshal([]byte(yamlString), &unmarshalledYaml)
-	if err != nil {
-		t.Errorf("error unmarshalling yaml: %v", err)
-	}
+	require.NoError(t, err)
 	return unmarshalledYaml
 }
```

Comment:
I think this function isn't necessary. It's only used once and using generics here complicates things. It would be simpler to remove it and handle it directly where it's called.

```golang
var untypedPkg T
err := goyaml.Unmarshal([]byte(badZarfPackage), &untypedPkg)
require.NoError(t, err)
```